addons/l10n_mx/controllers/portal.py

```diff
@@ -0,0 +1,31 @@
+from itertools import count
+
+from odoo.addons.portal.controllers import portal
+from odoo.http import request
+
+class CustomerPortal(portal.CustomerPortal):
+
+    def _get_mandatory_fields(self):
+
+        country_id = request.params.get('country_id')
```

Comment:
Actually you'll never get this if the country is not editable. So currently once you have 1 sale order, you can go back to your profile and remove the zip code.

So maybe we should pass the country id through context instead? (e.g. "portal_form_country_id") And then by param in master. As anyway getting it through the request is already a bit meh, don't think using context is that much worse.